template_feed/Microsoft.DotNet.Web.ProjectTemplates.2.0/content/WebApi-CSharp/Areas/IdentityService/Controllers/AccountController.cs

```diff
@@ -0,0 +1,419 @@
+﻿using System.Linq;
+using System.Security.Claims;
+using System.Threading.Tasks;
+using Microsoft.AspNetCore.Authentication;
+using Microsoft.AspNetCore.Authorization;
+using Microsoft.AspNetCore.Identity;
+using Microsoft.AspNetCore.Identity.Service;
+using Microsoft.AspNetCore.Mvc;
+using Microsoft.AspNetCore.Mvc.Rendering;
+using Microsoft.Extensions.Logging;
+using Company.WebApplication1.Identity.Models;
+using Company.WebApplication1.Identity.Models.AccountViewModels;
+using Company.WebApplication1.Identity.Services;
+
+namespace Company.WebApplication1.Identity.Controllers
+{
+    [Area(IdentityServiceConstants.Tenant)]
+    [Route(IdentityServiceConstants.RoutePrefix + "/[area]/[controller]/[action]")]
+    [Authorize(IdentityServiceOptions.LoginPolicyName), AllowAnonymous]
```

Comment:
They are both needed and highly related to each other. The Authorize attribute specifies how an authentication cookies should get handled (ex so the login partial works as expected). The AllowAnonymous attribute allows anonymous requests because all the actions on this controller support anonymous access. According to @davidfowl this is how it's done in the auth2 model. I'm happy to put them on separate lines if you want, but they both need to be there.